lib/livebook/hubs/enterprise_client.ex

```diff
@@ -97,4 +114,22 @@ defmodule Livebook.Hubs.EnterpriseClient do
   defp broadcast_message(message) do
     Phoenix.PubSub.broadcast(Livebook.PubSub, @pubsub_topic, message)
   end
+
+  defp registry_name(%Enterprise{url: url}) do
+    {:via, Registry, {@registry, url}}
+  end
+
+  defp put_secret(state, %Secret{name: name} = secret) do
+    secrets =
+      if Enum.any?(state.secrets, &(&1.name == name)) do
+        Enum.reduce(state.secrets, [], fn
+          %Secret{name: ^name}, acc -> [secret | acc]
+          item, acc -> [item | acc]
+        end)
+      else
+        [secret | state.secrets]
+      end
+
+    %{state | secrets: Enum.reverse(secrets)}
```

Comment:
There should be only one secret with the given name, right?

```suggestion
    secrets =
      if idx = Enum.find_index(state.secrets, &(&1.name == name)) do
        List.replace_at(state.secrets, idx, secret)
      else
        state.secrets ++ [secret]
      end

    %{state | secrets: secrets}
```

Reply:
If you connect to 2 enterprises, they could have the same secret name